src/Microsoft.Performance.SDK/Processing/ColumnBuilding/ColumnBuilder.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Performance.SDK.Processing.ColumnBuilding;
+
+/// <summary>
+///     Base class for classes that can build/extend upon columns that are
+///     part of a table.
+/// </summary>
+public abstract class ColumnBuilder
+{
+    internal ColumnBuilder()
+    {
+    }
+
+    /// <summary>
+    ///     Commits this builder's configuration to the column being built. Note that this method MUST be called to have
+    ///     any effect on the column being built. Calling this method multiple times will override previous calls.
```

Comment:
> Calling this method multiple times will override previous calls. [](http://example.com/codeflow?start=49&length=65)

I've been thinking about the flexibility of being able to call this more than once. Should we allow it? Would it better to throw if this has already been called?

It seems like the pattern we'd expect it to use the builder to build and build until the final tree is ready, then commit once.

Reply:
After discussing this offline, I have changed this API to
1. Make this method `internal`
2. Changing column building APIs to take in functions that return the final column builders they want to apply
3. Invoking Commit on the returned builders from within the runtime/engine